statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DefinitionProcessor.java

```diff
@@ -112,6 +113,11 @@ private static String fetchS3Definition(final S3Location s3Location, final Amazo
             throw new CfnInternalFailureException(e);
         }
 
+        // DefinitionSubstitution before validating JSON or YAML format
+        if (model.getDefinitionSubstitutions() != null) {
+            definition = transformDefinition(definition, model.getDefinitionSubstitutions());
```

Comment:
Question (blocking): does this still work if the S3 file is provided in YAML format?

Reply:
Yes, I tested both JSON and YAML format. I also have a unit test covering the YAML format, I'll make one for JSON in my next PR